src/java/org/apache/cassandra/service/accord/AccordConfigurationService.java

```diff
@@ -381,7 +388,7 @@ public synchronized void onNodesRemoved(long epoch, Topology current, Set<Node.I
     private long[] nonCompletedEpochsBefore(long max)
     {
         LongArrayList notComplete = new LongArrayList();
-        for (long epoch = epochs.minEpoch(); epoch <= max && epoch <= epochs.maxEpoch(); epoch++)
+        for (long epoch = epochs.minEpoch(), maxKnown = epochs.maxEpoch(); epoch <= max && epoch <= maxKnown; epoch++)
```

Comment:
I don't think this iteration is safe? `minEpoch` and `maxEpoch` can change while the loop is running. I think we need the epoch monitor when we run this loop.

Reply:
I do think we might due to the fact `accord.impl.AbstractConfigurationService.AbstractEpochHistory#truncateUntil` could be called and `accord.impl.AbstractConfigurationService.AbstractEpochHistory#getOrCreate` can re-populate the past epochs.

so added the lock